heartbeat/monitors/task.go

```diff
@@ -162,15 +162,20 @@ func runPublishJob(job jobs.Job, client beat.Client) []scheduler.TaskFunc {
 		}
 	}
 
-	if len(next) == 0 {
+	if !hasContinuations {
```

Comment:
I added this here since it makes the code more readable. I know this is a bug fix, but I think some small cleanup of this function goes to that since hard to maintain code carries bug risk.